library/src/scala/runtime/DynamicTuple.scala

```diff
@@ -180,6 +184,12 @@ object DynamicTuple {
     case self: Product => productToArray(self)
   }
 
+  def dynamicToIArray(self: Tuple): IArray[Object] = (self: Any) match {
+    case self: Unit => Array.emptyObjectArray.asInstanceOf[IArray[Object]]
+    case self: TupleXXL => self.elems
+    case self: Product => productToArray(self).asInstanceOf[IArray[Object]]
```

Comment:
It's better to create IArray through its constructors instead of cast.

Reply:
The array constructor would create a new empty array each time